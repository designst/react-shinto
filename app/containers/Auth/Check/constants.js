import createConstant from 'utils/createConstant';

export const AUTH_CHECK_REQUEST = createConstant(__filename, 'AUTH_CHECK_REQUEST');
export const AUTH_CHECK_SUCCESS = createConstant(__filename, 'AUTH_CHECK_SUCCESS');
export const AUTH_CHECK_FAILURE = createConstant(__filename, 'AUTH_CHECK_FAILURE');
