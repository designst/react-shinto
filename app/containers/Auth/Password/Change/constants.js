import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_CHANGE_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_REQUEST',
);
export const AUTH_PASSWORD_CHANGE_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_SUCCESS',
);
export const AUTH_PASSWORD_CHANGE_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_FAILURE',
);
