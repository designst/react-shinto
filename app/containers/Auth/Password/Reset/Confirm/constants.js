import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_RESET_CONFIRM_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_REQUEST',
);
export const AUTH_PASSWORD_RESET_CONFIRM_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_SUCCESS',
);
export const AUTH_PASSWORD_RESET_CONFIRM_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_FAILURE',
);
