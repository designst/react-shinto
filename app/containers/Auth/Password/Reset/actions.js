import {
  AUTH_PASSWORD_RESET_REQUEST,
  AUTH_PASSWORD_RESET_SUCCESS,
  AUTH_PASSWORD_RESET_FAILURE,
} from './constants';

export const passwordResetRequest = payload => ({
  type: AUTH_PASSWORD_RESET_REQUEST,
  payload,
});

export const passwordResetSuccess = data => ({
  type: AUTH_PASSWORD_RESET_SUCCESS,
  data,
});

export const passwordResetFailure = error => ({
  type: AUTH_PASSWORD_RESET_FAILURE,
  error,
});
