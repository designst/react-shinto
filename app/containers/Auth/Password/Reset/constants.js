import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_RESET_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_REQUEST',
);
export const AUTH_PASSWORD_RESET_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_SUCCESS',
);
export const AUTH_PASSWORD_RESET_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_FAILURE',
);
