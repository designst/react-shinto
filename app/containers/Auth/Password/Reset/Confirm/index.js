import { compose } from 'redux';
import { connect } from 'react-redux';

import PasswordResetConfirm from 'components/Auth/Password/Reset/Confirm';

import { passwordResetConfirmRequest } from './actions';

const mapStateToProps = (state, { match }) => ({
  initialValues: {
    token: match.params.token,
  },
});

const mapDispatchToProps = {
  onSubmit: passwordResetConfirmRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(PasswordResetConfirm);
