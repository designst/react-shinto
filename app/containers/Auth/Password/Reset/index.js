import { compose } from 'redux';
import { connect } from 'react-redux';

import PasswordReset from 'components/Auth/Password/Reset';

import { passwordResetRequest } from './actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSubmit: passwordResetRequest,
};

const connector = connect(mapStateToProps, mapDispatchToProps);

export default compose(connector)(PasswordReset);
