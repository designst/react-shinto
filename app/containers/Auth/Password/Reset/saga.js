import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_RESET_REQUEST } from 'containers/Auth/Password/Reset/constants';

import {
  passwordResetSuccess,
  passwordResetFailure,
} from 'containers/Auth/Password/Reset/actions';

const logger = createLogger(__filename);

const authPasswordResetUrl = process.env.SHINTO_AUTH_PASSWORD_RESET_API_ENDPOINT;

export function* authPasswordReset(action) {
  const { email, username } = action;

  const apiService = yield getContext('apiService');

  logger('Password Reset');

  try {
    const response = yield call(apiService.post, authPasswordResetUrl, {
      email,
      username,
    });
    yield put(passwordResetSuccess(response.data));
  } catch (err) {
    yield put(passwordResetFailure(err));
  }
}

export default function* authPasswordResetData() {
  yield takeLatest(AUTH_PASSWORD_RESET_REQUEST, authPasswordReset);
}
