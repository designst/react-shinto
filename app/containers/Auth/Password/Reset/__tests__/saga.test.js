import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import authPasswordResetData from 'containers/Auth/Password/Reset/saga';
import { AUTH_PASSWORD_RESET_REQUEST } from 'containers/Auth/Password/Reset/constants';

describe('authPasswordResetData', () => {
  let authPasswordResetGenerator;

  beforeEach(() => {
    authPasswordResetGenerator = authPasswordResetData();

    const selectDescriptor = authPasswordResetGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authPasswordResetSuccess action if it requests the data successfully', () => {
    const email = 'email@email.com';
    const username = 'username';

    return expectSaga(authPasswordResetData)
      .provide({
        getContext: () => ({
          post: url =>
            url === '/auth/password/reset'
              ? {
                  data: {
                    email,
                    username,
                  },
                }
              : {},
        }),
      })
      .dispatch({
        type: AUTH_PASSWORD_RESET_REQUEST,
        payload: {
          email,
          username,
        },
      })
      .withReducer(authReducer)
      .run();
  });

  it('should dispatch the authPasswordResetFailure action if it requests the data unsuccessfully', () => {
    const email = 'email@email.com';
    const username = 'username';

    return expectSaga(authPasswordResetData)
      .dispatch({
        type: AUTH_PASSWORD_RESET_REQUEST,
        payload: {
          email,
          username,
        },
      })
      .withReducer(authReducer)
      .run();
  });
});
