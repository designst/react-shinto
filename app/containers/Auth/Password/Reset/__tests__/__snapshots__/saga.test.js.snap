// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authPasswordResetData should dispatch the authPasswordResetFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/password/reset/constants/AUTH_PASSWORD_RESET_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`authPasswordResetData should dispatch the authPasswordResetSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/password/reset/constants/AUTH_PASSWORD_RESET_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
